service/proxy-service.py

```diff
@@ -38,52 +47,135 @@ def decorated(*args, **kwargs):
 @app.route('/file/<share_name>/<directory_name>/<file_name>', methods=['GET'])
 @requires_auth
 def get_file(share_name, directory_name, file_name):
-    auth = request.authorization
-    file_service = FileService(account_name = auth.username, account_key = auth.password)
-    f_stream = io.BytesIO()
     try:
+        account_name, account_key = get_auth(request.authorization)
+        file_service = FileService(account_name = account_name, account_key = account_key)
+        f_stream = io.BytesIO()
         file_service.get_file_to_stream(share_name, directory_name, file_name,
                                                f_stream, max_connections=6)
         f_stream.seek(0)
         return send_file(f_stream, attachment_filename=file_name, as_attachment=True)
     except Exception as e:
+        logger.exception(e)
+        return abort(500, e)
+
+
+def get_sas_params(args):
+    def str_to_timedelta(str):
+        days = 0
+        hours = 0
+        minutes = 0
+        if str:
+            str = str.upper()
+            if str[-1] == 'D':
+                days = int(str[:-1])
+            elif str[-1] == 'H':
+                hours = int(str[:-1])
+            elif str[-1] == 'M':
+                minutes = int(str[:-1])
+        return timedelta(days=days, hours=hours,minutes=minutes)
+    start_timedelta_in = args.get('start_timedelta', SAS_PARAMS.get('start_timedelta'))
+    expiry_timedelta_in = args.get('expiry_timedelta', SAS_PARAMS.get('expiry_timedelta'))
+    start_timedelta_out = str_to_timedelta(start_timedelta_in)
+    expiry_timedelta_out = str_to_timedelta(expiry_timedelta_in)
+    return start_timedelta_out, expiry_timedelta_out
+
+@app.route('/file/<share_name>/<directory_name>/<file_name>', methods=['POST'])
```

Comment:
This doesn't seem very robust. Will this support nested directories? This seems like it will only support files inside root level directories. I actually did something similar and met several issues. My plan is to rewrite that service using get params instead of relative url path. (haven't had time yet)